app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```
@@ -339,46 +349,50 @@ public void deleteSurvey(Survey survey) {
 
     @Override
     public void deleteSurveyPermanently(GuidCreatedOnVersionHolder keys) {
-        Survey existing = getSurvey(keys, false);
-        deleteAllElements(existing.getGuid(), existing.getCreatedOn());
-        surveyMapper.delete(existing);
         
-        // Delete the schemas as well, or they accumulate.
-        try {
-            StudyIdentifier studyId = new StudyIdentifierImpl(existing.getStudyIdentifier());
-            uploadSchemaService.deleteUploadSchemaById(studyId, existing.getIdentifier());
-        } catch(EntityNotFoundException e) {
-            // This is OK. Just means this survey wasn't published.
+        Survey existing = new QueryBuilder().setSurvey(keys.getGuid()).setCreatedOn(keys.getCreatedOn())
+                .setSkipElements(true).getOne(false);
+        if (existing != null) {
```

Unit test that deleting a non-existent survey fails silently?

---

Added